blazar/manager/service.py

```diff
@@ -446,53 +480,26 @@ def update_lease(self, lease_id, values):
             datetime.datetime.strftime(lease['end_date'], LEASE_DATE_FORMAT))
         before_end_date = values.get('before_end_date', None)
 
-        now = datetime.datetime.utcnow()
-        now = datetime.datetime(now.year,
-                                now.month,
-                                now.day,
-                                now.hour,
-                                now.minute)
-        if start_date == 'now':
-            start_date = now
-        else:
-            start_date = self._date_from_string(start_date)
-        if end_date == 'now':
-            end_date = now
-        else:
-            end_date = self._date_from_string(end_date)
+        start_date, end_date, now = self._parse_lease_dates(start_date,
+                                                            end_date)
 
         values['start_date'] = start_date
         values['end_date'] = end_date
 
-        if (lease['start_date'] < now and
-                values['start_date'] != lease['start_date']):
-            raise common_ex.InvalidInput(
-                'Cannot modify the start date of already started leases')
-
-        if (lease['start_date'] > now and
-                values['start_date'] < now):
-            raise common_ex.InvalidInput(
-                'Start date must be later than current date')
-
-        if lease['end_date'] < now:
-            raise common_ex.InvalidInput(
-                'Terminated leases can only be renamed')
-
-        if (values['end_date'] < now or
-           values['end_date'] < values['start_date']):
-            raise common_ex.InvalidInput(
-                'End date must be later than current and start date')
-
-        # NOTE(priteau): We call this function to check against the
-        # Chameleon lease duration policy
-        try:
-            values['user_id'] = lease['user_id']
-            values['project_id'] = lease['project_id']
-            self.usage_enforcer.check_lease_duration(values, lease=lease)
-        except exceptions.RedisConnectionError:
-            pass
+        self._check_for_invalid_date_inputs(lease, values, now)
 
         with trusts.create_ctx_from_trust(lease['trust_id']):
+            # NOTE(priteau): We call this function to check against the
+            # Chameleon lease duration policy
+            # NOTE(jakecoll): For some reason has stopped working while being
```

Comment:
That's very odd

Reply:
Yea, I'm not sure what to make of this. Blazar context is very finicky.

Reply:
Removed along with all legacy usage enforcement.